android/app/src/test/java/com/digginroom/digginroom/feature/room/customview/roominfoview/comment/CommentViewModelTest.kt

```diff
@@ -0,0 +1,123 @@
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.digginroom.digginroom.fixture.CommentFixture.Comment
+import com.digginroom.digginroom.fixture.CommentFixture.Comments
+import com.digginroom.digginroom.fixture.LogResult
+import com.digginroom.digginroom.model.mapper.CommentMapper.toModel
+import com.digginroom.digginroom.repository.CommentRepository
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class CommentViewModelTest {
+    private lateinit var commentViewModel: CommentViewModel
+    private lateinit var commentRepository: CommentRepository
+
+    @get:Rule
+    val instantExecutorRule = InstantTaskExecutorRule()
+
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(UnconfinedTestDispatcher())
+        commentRepository = mockk()
+        commentViewModel = CommentViewModel(commentRepository)
+    }
+
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun `해당 룸의 댓글 요청 시 성공하면 성공 상태와 함께 댓글을 받아온다`() {
+        // given
+        val defaultComments = Comments()
+        val roomId: Long = 0
+
+        coEvery {
+            commentRepository.findComments(any())
+        } returns LogResult.success(defaultComments)
+
+        // when
+        commentViewModel.findComments(roomId)
+
+        // then
+        val actual = commentViewModel.comments.value
+        val expected = defaultComments.map { it.toModel() }
+        assertEquals(actual, expected)
+    }
+
+    @Test
+    fun `댓글 작성을 요청하면 새 댓글이 생성된다`() {
+        // given
+        val defaultComment = Comment()
+        val roomId: Long = 0
+
+        coEvery {
+            commentRepository.postComment(any(), any())
+        } returns LogResult.success(defaultComment)
+
+        // when
+        commentViewModel.postComment(roomId, defaultComment.comment)
+
+        // then
+        val actual = commentViewModel.comments.value
+        val expected = listOf(defaultComment.toModel())
+        assertEquals(actual, expected)
+    }
+
+    @Test
+    fun `댓글 작성 요청에 실패하면 commentRequestState가 Failed 이 된다`() {
+        // given
+        val defaultComment = Comment()
+        val roomId: Long = 0
+
+        coEvery {
+            commentRepository.postComment(any(), any())
+        } returns LogResult.failure()
+
+        // when
+        commentViewModel.postComment(roomId, defaultComment.comment)
+
+        // then
+        val actual = commentViewModel.commentRequestState.value
+        assertTrue(actual is CommentRequestState.Failed)
```

Comment:
CommentRequestState.Failed가 어짜피 object 클래스이므로 다른 테스트 코드들과 같이 expected 변수를 CommentRequestState.Failed로 두고 assertEquals를 사용하면 더욱 일관성있는 테스트 코드가 되지 않을까요?

Reply:
CommentRequestState.Failed 가 실패 메시지를 담게 하기 위해서 data class 로 선언되어 있는 상태입니다!
따라서 class 타입으로 비교하기 위해 아래같이 적는 것은 어떻게 생각하시나요? 
```suggestion
      val actual = commentViewModel.commentRequestState.value ?: throw IllegalArgumentException()
        assertEquals(CommentRequestState.Failed::class, actual::class)
```

Reply:
아 Failed를 object 클래스로 착각했습니다.  기존에 작성하신 테스트 코드로 사용하는게 더 좋아보입니다! 혼동을 드려서 죄송합니다ㅜㅜ

Reply:
아닙니다!! ㅎㅎ 가독성 측면에서 고민이 됐었는데.. 기존 코드로 유지하겠습니다~!